tests/unit/src/kakarot/instructions/test_push_operations.cairo

```diff
@@ -40,6 +40,17 @@ func test__exec_push2_should_add_2_byte_to_stack{
     return ();
 }
 
+@external
+func test__exec_push2_should_not_add_2_byte_to_stack{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}() {
+    alloc_locals;
+    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(1, 0xFF);
+    let result = PushOperations.exec_push2(ctx);
+    TestHelpers.assert_stack_last_element_contains_uint256(result.stack, Uint256(0xFFFF, 0));
```

Comment:
this is probably never read, the test should fail at `PushOperations.exec_push2`

Reply:
actually for this task you may want to have instead one single test using `exec_push_i` and a `parametrize`, like:

```
@external
func test_exec_push_should_raise{
    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
}(i: felt) {
    alloc_locals;
    let ctx: model.ExecutionContext* = TestHelpers.init_context_with_bytecode(i - 1, 0xFF);
    PushOperations.exec_push_i(ctx, i);
}
```

and in the python file

```
@pytest.mark.parametrize("i", range(2, 32))
async def test_should_raise(i):
    with kakarot_error():
        await ...
```

Reply:
Yea with this we could test all of the push amounts. Much better approach, I like it.

Reply:
I'll look into abstracting the other existing tests and using a `parametrize` as well. Seems to me that there is a lot of repetition.

Reply:
@ClementWalter Let me know your thoughts, I added another commit which refactors the other tests. I also updated the screenshot in the description for this change.